### PR TITLE
Solver options

### PR DESCRIPTION
**Pull request by MaximeBouton**
This PR addresses #9 

- `create_policy` no longer exists
- `ValueIterationPolicy` has outer constructors rather than inner ones. 
- the solve method is `solve(solver, mdp)`
- the solver type has new fields: `verbose`, `include_Q` and `init_util`

### CODE REVIEW

File: src/vanilla.jl
Comment (rejuvyesh) [inline src/vanilla.jl]: Why not:

```julia
utility::Vector{Float64}=zeros(n_states(mdp)),
policy::Vector{Int64}=zeros(Int64, n_states(mdp)),
```

Then you won't need the extra if else.

Comment (MaximeBouton) [reply src/vanilla.jl]: Good point, I made the change.

**Review by zsunberg — APPROVED** [review]
Sorry for the delay - I was travelling, but thanks so much for doing this!!! This looks great! Switching to outer constructors makes my heart happy, haha! :heart: :heart:

**Comment by zsunberg** [comment]
Just thought of this - test QMDP against this before you merge it!!!

**Comment by MaximeBouton** [comment]
I am preparing a PR on QMDP. I will merge this one after

**Comment by MaximeBouton** [comment]
JuliaPOMDP/QMDP.jl#10

**Pull request merged**
